lib/schema_associations/active_record/associations.rb

```diff
@@ -90,6 +107,12 @@ def _load_schema_associations_associations
             _define_association(:belongs_to, foreign_key)
           end
 
+          (@schema_associations_deferred_associations || []).each do |a|
+            argstr = a[:args].inspect[1...-1] + ' # deferred association'
+            _create_association(a[:macro], a[:name], argstr, *a[:args])
+          end
+          @schema_associations_deferred_associations = nil
```

Comment:
Should we use `remove_instance_variable` instead?

Reply:
in looking through all of the Schema+ gems they all simply just assign to nil and none of them use remove_instance_variable.  And googling around there isn't much difference between the two for what we are trying to accomplish here. but I'll switch it to remove as that does more explicitly state what we are doing.